### PR TITLE
Final product

### PR DESCRIPTION
**Pull request by BANCUNGUYE66**
### In this PR i completed the following
 Splash screen
 
 * A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
 
 Sign up and log in pages
 
 * The user should be able to register in the app with full name, email and password (all mandatory).
 * The user can log into the app using email and password.
 * If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
 
 Home page (categories page)
 
 * When the user logs in, they are presented with the categories page.
 * For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
 * When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
 * There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
 
 Transactions page
 
 * For a given category, the list of transactions is presented, ordered by the most recent.
 * At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
 * There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
 * When the user clicks on the "Back" button (<), the user navigates to the home page.
 
 "Add a new category" page

 * The user fills out a form to create a new category, indicating their name and icon (both mandatory).
 * The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
 * When the user clicks on the "Back" button (<), the user navigates to the home page.
 
 "Add a new transaction" page
 
  The user fills out a form to create a new transaction with:
 
 * name (mandatory)
 * amount (mandatory)
 * categories (mandatory at least one)
 
 The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
 
 When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
 
 Testing requirements
 
 * Create unit and integration tests for all the most important components of your RoR application.

### CODE REVIEW

**Review by HAMMAS-SALEEM — COMMENTED** [review]
# Invalid Submission :red_circle:

invalid-submission

## Hey @BANCUNGUYE66 :wave:

Good job on implementing some of the requirements but I cannot proceed review of your pull request because there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

# Mandatory Changes ♻️

- [x] It would be nice if you fix the migrations because I am not able to run the migrations 😨 Kindly fix them and submit this pull request again 👍

**Comment by BANCUNGUYE66** [comment]
@HAMMAS-SALEEM Thank you for your review time, but in my project, everything is working well as you can check in the presentation video in the README file. 
Screenshot 2023-09-14 at 21 29 10

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      <%= f.password_field :password, autocomplete: "current-password", placeholder: "Password"  %>
+    </div>
+
+    <div class="actions">
+      <%= f.submit "Log in", id:"submit-btn" %>
+    </div>
+  <% end %>
+
+  <%= render "devise/shared/links" %>
+</div>
```
Comment (Munsa1) [inline app/views/devise/sessions/new.html.erb:17]: - [ ] Good job on the mobile design, however, we need to follow the given design for the Login and Sign up pages, including the colors, layout, and typography.


Your design
image

image

image


Given design
image

image

image


- Kindly fix the color of the button in your design
- Also kindly follow the given layout in your design
- Kindly use the correct font family for your design, you have used `Poppins` whilst the given font is `Proxima Nova Bold`

### Note that the font family layout and design applies to the entire application

File: README.md [hunk 2]
```diff
@@ -1,24 +1,149 @@
...
+- User can login
+- User can create a transaction
+- user can create a category
+- user can view all transactions based on category
 
-* Deployment instructions
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-* ...
+## 🚀 Live Demo <a name="live-demo" ></a>
```
Comment (Munsa1) [inline README.md:55]: - [ ] It is a requirement that you deploy your app using Render, GitHub pages, or Netlify, kindly do so and add the live link to the `Live Demo` section of your readme file

File: app/views/entities/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                <%= image_tag('calendar.png') %> <%= entity.created_at %>
+              </h3>
+          </li>
+      <% end %>
+    </ul>
+  </div>
+  <%= link_to({ controller: 'entities', action: 'new' }) do %>
+    <button>Add new transaction</button>
+  <%end%>
+</div>
```
Comment (Munsa1) [inline app/views/entities/index.html.erb]: - [x] Kindly follow the given design for this page as well as per requirements.

Your design
image

Given design
image

- Kindly ensure you follow the layout of elements on your page to match the given design

File: db/migrate/20230912104205_create_groups.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class CreateGroups < ActiveRecord::Migration[7.0]
+  def change
+    create_table :groups do |t|
+      t.string :name
+      t.string :icon
+      t.references :author, null: false, foreign_key: { to_table: :users }
```
Comment (Munsa1) [inline db/migrate/20230912104205_create_groups.rb:6]: - [ ] Kindly add indexes for all your foreign keys as per requirement

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @BANCUNGUYE66, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>
## Highlights
Good Job On The Video Presentation :heavy_check_mark: 
Used Devise For Authentication :heavy_check_mark: 

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by Munsa1** [comment]
- [ ] Kindly add an MIT License Agreement to your readme file, for a more professional readme file
image

- [ ] Also ensure that the issues link points to this project's issues page and not to another capstone under the Contributing section of your readme file
image

image


- [ ] Lastly, kindly fix the link to your `LinkedIn` profile under the Author section of your readme file as it is a broken link leading nowhere
image

**Comment by Munsa1** [comment]
- [ ] Kindly fix the broken link to the `Tech Stack` under the built-with section by adding a subheading so that when the user clicks on this link they get taken to the tech stack. This will make the readme file more professional.
image

Clicking on the link above should take the user to the part below
image

- [ ] Also consider fixing the following broken links in the table of contents pointing to subsections of the `Getting Started` section. Clicking on the `Usage` and `Deployment` links should take the user to the Usage and Deployment parts of the Getting Started section respectively
image
image

**Review by Alfredbis29 — COMMENTED** [review]
Hi @BANCUNGUYE66  

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has already the request changes and i\on my sdie is the first  review, kindly open another PR and request for another review

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request closed without merging**